datasetinsights/estimators/faster_rcnn.py

```diff
@@ -306,20 +305,24 @@ def train_one_epoch(
                     f"(total training examples: {examples_seen}) is "
                     f"{intermediate_loss}"
                 )
-                self.writer.add_scalar(
-                    "training/intermediate_loss",
-                    intermediate_loss,
-                    examples_seen,
-                )
+                if is_master():
+                    self.writer.add_scalar(
+                        "training/intermediate_loss",
+                        intermediate_loss,
+                        examples_seen,
+                    )
             losses_grad.backward()
             if (i + 1) % accumulation_steps == 0:
                 optimizer.step()
                 lr_scheduler.step()
                 optimizer.zero_grad()
-        self.writer.add_scalar("training/loss", loss_metric.compute(), epoch)
-        self.writer.add_scalar(
-            "training/lr", optimizer.param_groups[0]["lr"], epoch
-        )
+        if is_master():
```

Comment:
Where should we decide which file writer should be used, during estimator initialization?

Reply:
What do you think? What's more convenient for users who tries to create a new estimator?

Reply:
I think it'll be more convenient for user to initialize in the estimator itself.

Reply:
What do you mean by "for user to initialize in the estimator itself"? It might be easier to have this in `create_estimator` so that every Estimator subclass has this by default. They can override this if they want in their own constructor.